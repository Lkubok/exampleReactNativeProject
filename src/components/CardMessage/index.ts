import Component from './CardMessage';

export default Component;
