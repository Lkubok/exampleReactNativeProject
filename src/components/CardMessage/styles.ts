import { StyleSheet } from 'react-native';

import { variables } from 'res';

export default StyleSheet.create({
  cardDataRight: {
    alignItems: 'center',
    flexDirection: 'row',
    flexWrap: 'nowrap',
  },
  cardDataWrapper: {
    flex: 1,
  },
  cardImage: {
    borderRadius: variables.borderRadius.min,
    height: 46,
    width: 46,
  },
  cardImageWrapper: {
    flexShrink: 0,
    marginRight: variables.spacing.medium,
    width: 46,
  },
  cardTextWrapper: {
    paddingTop: variables.spacing.base,
  },
  cardTopWrapper: {
    flexDirection: 'row',
  },
  containerStyle: {
    backgroundColor: variables.colors.white,
    borderRadius: variables.borderRadius.min,
    elevation: 3,
    flexDirection: 'column',
    marginTop: variables.spacing.medium,
    paddingHorizontal: variables.spacing.medium,
    paddingVertical: variables.spacing.medium,
    shadowColor: variables.colors.black,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.2,
    shadowRadius: 2,
  },
  nameTextWrapper: {
    alignSelf: 'baseline',
    flexGrow: 1,
    flexShrink: 1,
  },
  nameWrapper: { alignItems: 'flex-start', flexDirection: 'row', marginBottom: variables.spacing.min },
  unreadDot: {
    alignItems: 'baseline',
    backgroundColor: variables.colors.accentGreen,
    borderRadius: 4,
    flexShrink: 0,
    height: 8,
    marginLeft: variables.spacing.small,
    width: 8,
  },
});
