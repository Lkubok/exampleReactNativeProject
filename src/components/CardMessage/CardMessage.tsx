import Typography from 'components/Typography';
import React, { FC } from 'react';
import { Image, View } from 'react-native';
import dayjs from 'dayjs';

import { useNavigation } from '@react-navigation/native';
import { screenNames } from 'navigation';

import styles from './styles';
import { variables } from 'res';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { useDispatch } from 'react-redux';
import { actions as offerActions } from 'store/offers';

export type CardMessageParamsProps = {
  offer_slug: string;
  customer_slug: string;
  recipientSlug: string;
  offerName: string;
};

interface CardMessageProps {
  imageUrl: string | undefined;
  name: string;
  time: string;
  productName: string;
  unread: boolean;
  message: string;
  recipientName: string;
  recipientPhoto: string;
  recipientVerified: boolean;
  recipientCompanyName: string;
  params?: CardMessageParamsProps;
}

export interface Props {
  item: CardMessageProps;
}

const CardMessage: FC<Props> = ({ item }) => {
  const { navigate } = useNavigation();
  const { time, recipientName, recipientPhoto } = item;
  const dispatch = useDispatch();
  const formattedDate = dayjs(time).format('DD.MM.YYYY');

  return (
    <TouchableOpacity
      style={styles.containerStyle}
      onPress={() => {
        dispatch(offerActions.setCurrentOfferData(null));

        navigate(screenNames.MainStackNames.NewsStack, {
          screen: screenNames.MainFlow.NewsStack.NewsDetail,
          params: {
            ...item.params,
            recipientName: item.recipientName,
            recipientPhoto: item.recipientPhoto,
            recipientCompanyName: item.recipientCompanyName,
            recipientVerified: item.recipientVerified,
          },
        });
      }}>
      <View style={styles.cardTopWrapper}>
        {recipientPhoto && (
          <View style={styles.cardImageWrapper}>
            <Image source={{ uri: recipientPhoto }} style={styles.cardImage} />
            {/* TODO: ADD VERIFIED BADGE */}
          </View>
        )}
        <View style={styles.cardDataWrapper}>
          <View style={styles.nameWrapper}>
            <View style={styles.nameTextWrapper}>
              <Typography type="subheading" text={recipientName} />
            </View>
            <View style={styles.cardDataRight}>
              <Typography type="text-muted" text={formattedDate} />
              {item.unread && <View style={styles.unreadDot} />}
            </View>
          </View>
          <Typography color={variables.colors.accentGreen} type="small" text={item.productName} />
        </View>
      </View>
      <View style={styles.cardTextWrapper}>
        <Typography bold={item.unread} type="text" text={item.message} />
      </View>
    </TouchableOpacity>
  );
};

export default CardMessage;
