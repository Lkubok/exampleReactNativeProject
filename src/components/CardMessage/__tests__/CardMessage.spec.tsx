import React from 'react';
import { act, customRender } from 'utils/testUtils';

import CardMessage, { Props } from '../CardMessage';

jest.useFakeTimers();

describe('CardMessage component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: Props;
  beforeEach(() => {
    props = {
      item: {
        imageUrl: '',
        message: '',
        name: '',
        productName: '',
        recipientCompanyName: '',
        recipientName: '',
        recipientPhoto: '',
        recipientVerified: true,
        time: '',
        unread: false,
        params: { customer_slug: 'd', offerName: '', offer_slug: '', recipientSlug: '' },
      },
    };
  });

  it('should match snapshot when 1 line', () => {
    const snapshot = customRender(<CardMessage {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
