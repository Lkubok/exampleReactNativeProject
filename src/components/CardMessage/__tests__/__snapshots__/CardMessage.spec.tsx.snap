// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMessage component should match snapshot when 1 line 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
  >
    <View
      accessible={true}
      style={
        Object {
          "backgroundColor": "[7m[38;2;255;255;255m#ffffff[39m[27m",
          "borderRadius": 6,
          "elevation": 3,
          "flexDirection": "column",
          "marginTop": 16,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 16,
          "shadowColor": "[7m[38;2;0;0;0m#000000[39m[27m",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginBottom": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "baseline",
                  "flexGrow": 1,
                  "flexShrink": 1,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "black",
                      "fontFamily": "SFCompactText-Semibold",
                      "fontSize": 18,
                      "letterSpacing": -0.36,
                    }
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "nowrap",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "[7m[38;2;138;138;142m#8a8a8e[39m[27m",
                    "fontFamily": "SFCompactText-Medium",
                    "fontSize": 16,
                    "letterSpacing": -0.32,
                  }
                }
              >
                Invalid Date
              </Text>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "[7m[38;2;73;158;85m#499e55[39m[27m",
                  "fontFamily": "SFCompactText-Medium",
                  "fontSize": 16,
                  "letterSpacing": -0.32,
                }
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingTop": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 17,
              "letterSpacing": -0.34,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>
</RNCSafeAreaProvider>
`;
